src/main.rs

```diff
@@ -45,6 +52,58 @@ impl<W: std::io::Write> RockPaperScissorsGame<W>{
       _ => ShapesInput::Unrecognized
     }
   }
+
+  fn play_round(&mut self) -> Result<bool, std::io::Error> {
```

Comment:
and here we go. V principe teraz si si pripravil potrebne struktury a teraz ides implementovat game logic a vsetko k tomu, aby sa to dalo spustit. To je v principe presne obratene od toho, ako by to malo byt. Ty robis
```
nehratelne + nova funkcionalita -> nehratelne + refaktor -> nehratelne + nova funkcionalita -> nehratelne + refaktor ... -> hratelne
```
Pointa toho iterativneho robenia je, ze ty chces byt schopny chytat bugy a problemy cely cas. Ty ich takto vies chytat iba odvtedy, ako je to hratelne. Pozri sa na to ako som to spravil ja
```
hratelne a blbo napisane -> hratelne + refaktor -> hratelne + refaktor -> hratelne + nova funkcionalita ...
```
atd. Na tvojom mieste by som zacal s tym, ze by sa to hrat a vsetka logika by bola v `main`. Dovod je, ze takemu kodu vsetci rozumeju, ako prvy commit to dava uplne zmysel. A postupne moze ist logika z mainu von. Tak ako si to spravil teraz, keby som netusil o co ti vlastne ide, tak vobec nerozumiem o co sa vobec snazis. Tvoj intention je skryty az po commit c.6 alebo kolko.

Iterativny development nema byt o tom, ze si spravis commit vzdy ked ti to dava zmysel. Iterativny development je o tom, ze spravis commit vzdy ked to citatelovi dava zmysel. Nerobis to hlavne kvoli sebe, ale kvoli reviewerom.

Odporucam ti ako cvicenie to skusit prerobit takymto sposobom. Ziadne shapes, ziadny game logic, blby dvojito vnoreny `match` o tom, ze ci si vyhral alebo prehral, that's it.